Sources/ArcGISToolkit/Components/Augmented Reality/Utilities/GeoTrackingSceneView.swift

```diff
@@ -0,0 +1,199 @@
+    @State private var coachingOverlayIsActive = false
+    /// The current device heading.
+    @State private var currentHeading: Double?
+    /// The current device location.
+    @State private var currentLocation: Location?
+    /// The current interface orientation.
+    @State private var interfaceOrientation: InterfaceOrientation?
+    
+    /// Creates a world scale geo-tracking scene view.
+    /// - Parameters:
+    ///   - calibrationViewModel: The view model for accessing the calibration values.
+    ///   - clippingDistance: Determines the clipping distance in meters around the camera. A value
+    ///   of `nil` means that no data will be clipped.
+    ///   - initialCameraIsSet: A Boolean value that indicates whether the initial camera is set for the scene view.
+    ///   - calibrationViewIsPresented: A Boolean value that indicates whether the calibration view is present.
+    ///   - locationDataSource: The location datasource used to acquire the device's location.
+    ///   - sceneView: A closure that builds the scene view to be overlayed on top of the
+    ///   augmented reality video feed.
+    public init(
+        calibrationViewModel: WorldScaleSceneView.CalibrationViewModel,
+        clippingDistance: Double?,
+        initialCameraIsSet: Binding<Bool>,
+        calibrationViewIsPresented: Bool,
+        locationDataSource: LocationDataSource,
+        @ViewBuilder sceneView: @escaping (SceneViewProxy) -> SceneView
+    ) {
+        self.calibrationViewModel = calibrationViewModel
+        _initialCameraIsSet = initialCameraIsSet
+        self.calibrationViewIsPresented = calibrationViewIsPresented
+        self.locationDataSource = locationDataSource
+        
+        sceneViewBuilder = sceneView
+        
+        let cameraController = TransformationMatrixCameraController()
+        cameraController.translationFactor = 1
+        cameraController.clippingDistance = clippingDistance
+        _cameraController = .init(initialValue: cameraController)
+    }
+    
+    public var body: some View {
+        SceneViewReader { sceneViewProxy in
+            ZStack {
+                ARSwiftUIView(proxy: arViewProxy)
+                    .onDidUpdateFrame { _, frame in
+                        guard let interfaceOrientation, initialCameraIsSet else { return }
+                        
+                        sceneViewProxy.updateCamera(
+                            frame: frame,
+                            cameraController: cameraController,
+                            orientation: interfaceOrientation,
+                            initialTransformation: .identity
+                        )
+                        sceneViewProxy.setFieldOfView(
+                            for: frame,
+                            orientation: interfaceOrientation
+                        )
+                    }
+                    .onDidChangeGeoTrackingStatus { session, status in
+                        switch status.state {
```

Comment:
I wonder if everything in this closure should go in a @MainActor method called something like "handleGeoTrackingStatusChange". Because I see that `initialCameraIsSet` must be set on the mainthread, but I am also seeing that it's being read on line 104, and that probably should be on the same thread as well.

Reply:
I feel it's more like a timing issue. Will leave it to more testing. 🤔

Reply:
issue 4991